File: Army/state/wolfstate.cpp
#include "wolfstate.h"

using namespace std;

WolfState::WolfState(Unit* currentStateUnit) : State(currentStateUnit) {
    this->healthPointsLimit = currentStateUnit->getHPLimit() * 2;
    this->currentHP = currentStateUnit->getCurrentHP() * 2;
    this->damage = currentStateUnit->getDamage() * 2;
}

WolfState::~WolfState() {}

void WolfState::takeMagicDamage(int damage) {
    int newHP = currentStateUnit->getCurrentHP();
    int newDamage = damage * 2;    

    if ( currentHP <= 0 ) {
        currentStateUnit->setCurrentHP(0);

        return;
    }
    
    currentStateUnit->setCurrentHP(newHP - newDamage);
}

File: Army-Java/State/WolfState.java
package com.gymfox.Army.State;

import com.gymfox.Army.Units.Unit;

public class WolfState extends State {
    public WolfState(Unit currentUnitState) {
        super(currentUnitState);
        setStateName("Wolf");
        setHealthPointLimit(currentUnitState.getHealthPointLimit() * 2);
        setDamage(currentUnitState.getDamage() * 2);
    }

    @Override
    public void takeMagicDamage(int damage) {
        int newDamage = damage * 2;

        tryKill(newDamage);
        currentUnitState.setCurrentHP(currentUnitState.getCurrentHP() - newDamage);
    }
}
